pylint/checkers/utils.py

```diff
@@ -803,3 +806,29 @@ def node_type(node):
     except astroid.InferenceError:
         return
     return types.pop() if types else None
+
+
+def is_registered_in_singledispatch_function(node):
+    if not isinstance(node, astroid.FunctionDef):
+        return False
+
+    decorators = node.decorators.nodes if node.decorators else []
+    for decorator in decorators:
+        # func.register are function calls
+        if not isinstance(decorator, astroid.Call):
+            continue
+
+        func = decorator.func
+        if not isinstance(func, astroid.Attribute) or func.attrname != 'register':
+            continue
+
+        try:
+            func_def = next(func.expr.infer())
+        except (astroid.InferenceError, StopIteration):
```

Comment:
This should not raise StopIteration. It it does, it should be considered a bug in astroid in fixed.